test/source/tests/setup.ts

```diff
@@ -643,42 +654,42 @@ AN8G3r5Htj8olot+jm9mIa5XLXWzMNUZgg==
       }
       await gmailPage.waitTillGone('@dialog-passphrase');
       await PageRecipe.waitForToastToAppearAndDisappear(gmailPage, 'Account keys updated');
-      const set7 = await retrieveAndCheckKeys(1);
+      const set7 = await retrieveAndCheckKeys(settingsPage, acct, 1);
       expect(set7[0].lastModified!).to.be.greaterThan(set6[0].lastModified!); // an update happened
       await gmailPage.close();
       // 7. EKM returns an older version of the existing key, no toast, no update
-      MOCK_KM_UPDATING_KEY.response = { privateKeys: [{ decryptedPrivateKey: someOlderVersion }] };
+      MOCK_KM_UPDATING_KEY[acct].response = { privateKeys: [{ decryptedPrivateKey: someOlderVersion }] };
       gmailPage = await browser.newPage(t, TestUrls.mockGmailUrl(), undefined, extraAuthHeaders);
       await PageRecipe.noToastAppears(gmailPage);
       await gmailPage.notPresent('@dialog-passphrase');
-      const set8 = await retrieveAndCheckKeys(1);
+      const set8 = await retrieveAndCheckKeys(settingsPage, acct, 1);
       expect(set8[0].lastModified).to.equal(set7[0].lastModified); // no update
       await gmailPage.close();
       // 8. EKM returns an older version of the existing key, and a new key, toast, new key gets added encrypted with the same passphrase
-      MOCK_KM_UPDATING_KEY.response = { privateKeys: [{ decryptedPrivateKey: someOlderVersion }, { decryptedPrivateKey: testConstants.existingPrv }] };
+      MOCK_KM_UPDATING_KEY[acct].response = { privateKeys: [{ decryptedPrivateKey: someOlderVersion }, { decryptedPrivateKey: testConstants.existingPrv }] };
       gmailPage = await browser.newPage(t, TestUrls.mockGmailUrl(), undefined, extraAuthHeaders);
       await PageRecipe.waitForToastToAppearAndDisappear(gmailPage, 'Account keys updated');
       await gmailPage.notPresent('@dialog-passphrase');
-      const set9 = await retrieveAndCheckKeys(2);
+      const set9 = await retrieveAndCheckKeys(settingsPage, acct, 2);
```

Comment:
@tomholub 
I didn't test earlier that the newly-added keys (with a found or entered passphrase) are immediately ready-to-use ("unlocked" in a sense that their passphrase is stored in either cache or local storage). So this test is failing now.
What would be our strategy on saving passphrases for new keys (we don't provide the checkbox for the user unless the setup page scenario is used)? Should the passphrase for the newly-added key be saved to storage if the existing key (that was used to provide the passphrase) had its passphrase in the storage (settings inheritance)? Or some straightforward algo based on EKM flags (PASS_PHRASE_QUIET_AUTOGEN, DEFAULT_REMEMBER_PASS_PHRASE, FORBID_STORING_PASS_PHRASE...)?
Also, updating keys from EKM may silently extend passphrase cache expiration (4 hours) in session memory if we simply make a copy of the passphrase for the new longid. Should we care about it?

Reply:
ping @tomholub 
I can fix this in this PR, should I? If so, please confirm the strategy.

Reply:
Well noticed.

> Should the passphrase for the newly-added key be saved to storage if the existing key (that was used to provide the passphrase) had its passphrase in the storage (settings inheritance)?

Yes, that is the most reasonable

>  Also, updating keys from EKM may silently extend passphrase cache expiration (4 hours) in session memory if we simply make a copy of the passphrase for the new longid. Should we care about it?

I find this ok, as long as it only happens when there is an actual change of the keys. Wouldn't want this to happen each time EKM is checked for new keys with no updates to process.

Let's do that in this PR. Thanks!

Reply:
Done